File: 프로그래머스/greedy/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    int lost_number = lost.size();
    int reserve_number = reserve.size();
    for(int i = 0; i < lost.size(); i++){
        
        for(int j = 0; j < reserve.size(); j++){
            if(lost[i] == reserve[j]){
                lost[i] = 0;
                reserve[j] = -1;
                lost_number -= 1;
                reserve_number -= 1;
                break;
            }
        }
        
    }
    
    for(int i = 0; i < lost.size(); i++){
        if(lost_number == 0 || reserve_number == 0) break;
        if(lost[i] == 0 )continue;
        
        for(int j = 0; j < reserve.size(); j++){
            if(lost[i] - 1 == reserve[j]){
                lost[i] = 0;
                reserve[j] = -1;
                lost_number -=1;
                reserve_number -=1;
                break;
            }
            else if(lost[i] + 1 == reserve[j]){
                lost[i] = 0;
                reserve[j] = -1;
                lost_number -= 1;
                reserve_number -= 1;
                break;
            }
        }
        
        
        
        
        
    }
    
    
    
    
    
    int count = 0;
    for(int i = 0; i<lost.size();i++){
        if(lost[i] != 0)count++;
    }
    answer = n - count;
    

    return answer;
}

File: 프로그래머스_python/연습문제/Level1/체육복.py
def solution(n, lost, reserve):
    answer = 0
    
    lost.sort()
    reserve.sort()
    
    lost_copy = lost[:]
    
    
    for i in range(len(lost)):
        
        if lost[i] in reserve:
            lost_copy.remove(lost[i])
            reserve.remove(lost[i])
            
        elif (lost[i] - 1) in reserve:
            
            lost_copy.remove(lost[i])
            reserve.remove(lost[i] - 1)
            
        elif (lost[i] + 1) in reserve and (lost[i] + 1) not in lost:
            lost_copy.remove(lost[i])
            reserve.remove(lost[i] + 1)
            
            
        
    answer = n - len(lost_copy)
    
    return answer
